e2e/cypress/integration/channel_sidebar/DM_order_spec.js

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Stage: @prod
+// Group: @channel_sidebar
+
+import {getAdminAccount} from '../../support/env';
+
+describe('Channel sidebar', () => {
+    const sysadmin = getAdminAccount();
+    let testUser;
+    before(() => {
+        // # Enable channel sidebar organization
+        cy.apiUpdateConfig({
+            ServiceSettings: {
+                ExperimentalChannelSidebarOrganization: 'default_on',
+            },
+        });
+
+        // # Login as test user and visit town-square
+        cy.apiInitSetup({loginAfter: true}).then(({team, user}) => {
+            testUser = user;
+            cy.visit(`/${team.name}/channels/town-square`);
+
+            // # upgrade user to sys admin role
+            cy.externalRequest({user: sysadmin, method: 'put', path: `users/${user.id}/roles`, data: {roles: 'system_user system_admin'}});
+        });
+
+        // # Close "What's new" modal
+        cy.uiCloseWhatsNewModal();
+    });
+
+    it('should order DMs based on the sort button', () => {
+        const usersPrefixes = ['a', 'c', 'd', 'j', 'p', 'u', 'x', 'z'];
+        const usernames = [];
+        usersPrefixes.forEach((prefix) => {
+            // # Create users with prefixes in alphabatical order
+            cy.apiCreateUser({prefix}).then(({user: newUser}) => {
+                cy.apiCreateDirectChannel([testUser.id, newUser.id]).then((res) => {
+                    const channel = res.body;
+
+                    // # Post messaege in The DM channel
+                    cy.postMessageAs({sender: newUser, message: 'test', channelId: channel.id});
+
+                    // add usernames in array for reference
+                    usernames.push(newUser.username);
+                });
+            });
+        });
+
+        // get DM category group
+        cy.get('button[aria-label="DIRECT MESSAGES"]').parents('.SidebarChannelGroup').within(() => {
+            const usernamesReversed = usernames.reverse();
+
```

Comment:
I've got the failing test on last user 'cause it didn't reflect to DM list in time.
<img width="569" alt="Screen Shot 2020-07-27 at 7 18 57 PM" src="https://user-images.githubusercontent.com/5334504/88537789-c46a3300-d040-11ea-9350-55fd8d990355.png">

Can we add some wait mechanism to DMs in the sidebar before it continue? Like below,
```javascript
cy.waitUntil(() => cy.get('.NavGroupContent').children().then((el) => {
    return el.length === usernames.length;
}));
```

Reply:
its another issue, I was reversing the array so, it changes the original copy causing issues

Reply:
They should already exist in the UI by the time this step executes